crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -23,6 +23,13 @@ def status
         ::Crowbar::UpgradeStatus.new.progress
       end
 
+      def node_status
+        {
+          upgraded: ::Node.all.reject(&:admin?).select(&:ready_after_upgrade?).map(&:name),
+          not_upgraded: ::Node.find("state:crowbar_upgrade").map(&:name)
```

Comment:
what if something gets out-of-sync somehow? iirc I had some cases where node switched to 'no-data' and then when it returned it was 'ready' not 'upgrade'... but I might be wrong.